en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md

```diff
@@ -0,0 +1,341 @@
+        them. You can write the new implementation by extending the 
+        `            AbstractValidator           `  in Apache Amber
+        component.
+2.  When implementation is done, package your class as a .jar file and
+    place it in the
+    `          <IS_HOME>/repository/component/lib         ` directory.
+3.  To register the custom grant type, configure the
+    `           <IS_HOME>/repository/conf/deployment.toml          `
+    file by adding a new entry, in a manner similar to the following
+    example;
+
+    ```toml
+    [[oauth.custom_grant_type]]
+    name="grant type identifier"
+    grant_handler="full qualified class name of grant handler"
+    grant_validator="full qualified class name of grant validator"
+    [oauth.custom_grant_type.properties]
+    IdTokenAllowed=true
+    ```
+
+    !!! info 
+        Setting the `            <IdTokenAllowed>           ` parameter to
+        `            true           `, provides flexibility to control the
+        issuing of IDtoken for each grant, and also allows the OIDC scope
+        validator to validate the grant types that should support the openid
+        scope.
+
+4. Restart the server to apply changes.
+
+To test this out, follow the instructions below to implement a custom-grant type sample.
+
+------------------------------------------------------------------------
+
+## Use the grant type sample
+
+The sample demonstrated here defines a new sample grant type called the
+"mobile" grant type. It is similar to the password grant type and the
+only difference is that a mobile number will be passed through instead
+of a password. The request to the `         /token        ` API must
+contain the following two request parameters.
+
+-   `          grant_type=mobile         `
+-   `          mobileNumber=044322433         `
+
+You can access the new grant type project sample from
+[here](https://github.com/wso2/samples-is/tree/v4.5.6/oauth2/custom-grant)
+. The grant handler and validator class is inside the
+`         org.wso2.sample.identity.oauth2.grant.mobile        ` package.
+This can be modified as required.
+
+### Configure the resource
+
+The following Maven buildable source is compatible with WSO2 Identity
+Server 5.1.0 onwards. The attached `          .jar         ` file can be
+directly used as well.
+
+| Buildable source   | [custom-grant](https://github.com/wso2/samples-is/tree/v4.5.6/oauth2/custom-grant) |
+|--------------------|------------------------------------------------------------------------------------|
+| Built jar file | [custom-grant-4.5.6.jar]({{base_path}}/assets/attachments/custom-grant-4.5.6.jar)       |
+
+!!! note
+    To generate the .jar file from the buildable source, run the following Apache Maven
+    command in the sample's location using the command line.
+
+    ``` java
+    mvn clean install
+    ```
+    
+2.  Copy the .jar file in target directory into the
+    `          <IS_HOME>/repository/component/lib         `
```

Comment:
same comment https://github.com/wso2/docs-is/pull/4268/files#r1432582017 applies here as well

Reply:
Same as https://github.com/wso2/docs-is/pull/4268#discussion_r1433443588